generator/src/test/java/com/scottlogic/deg/generator/cucumber/Random.feature

```diff
@@ -0,0 +1,145 @@
+
+Feature: User can generate valid data for all types (string, numeric or temporal) in random generation mode. Actual randomness of the data not tested.
+
+  Background:
+    Given the generation strategy is random
+      And there is a field foo
+
+Scenario: The generator produces valid 'Temporal' data in random mode
+  Given foo is of type "temporal"
+    And foo is anything but null
+    And the generator can generate at most 5 rows
+    And foo is before or at 2019-01-01T00:00:00.000
+  Then 5 rows of data are generated
+    And foo contains temporal data
+    And foo contains anything but null
+    And foo contains temporal values between 0001-01-01T00:00:00.000 and 2019-01-01T00:00:00.000 inclusively
+
+Scenario: The generator produces valid 'Numeric' data in random mode
+  Given foo is of type "numeric"
+    And foo is anything but null
+    And the generator can generate at most 5 rows
+    And foo is less than or equal to 10
+  Then 5 rows of data are generated
+    And foo contains numeric data
+    And foo contains numeric values between -179769313486231570000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000 and 10 inclusively
```

Comment:
I am a little concerned by this value - this is an approximation of `-Double.Max_Value` and so there is the chance that the generated value will be outside of this range. @sl-slaing Can we look at introducing a value replacement for this so that the test would look like:
>And foo contains numeric values between minimum_number and 10 inclusively

Reply:
The following code can be used to calculate the minimum possible value:
```java
BigDecimal.valueOf(2).pow(1024).negate().subtract(BigDecimal.ONE)
```

This code presents the minimum value as
```
-179769313486231590772930519078902473361797697894230657273430081157732675805500963132708477322407536021120113879871393357658789768814416622492847430639474124377767893424865485276302219601246094119453082952085005768838150682342462881473913110540827237163350510684586298239947245938479716304835356329624224137217
```

Reply:
Suggest using `{field} contains numeric values less than or equal to {value}` step to remove the need to include the minimum possible decimal/numeric value